### PR TITLE
feat(medusa,dashboard): Tax settings pages and fixes to list tax rates endpoint

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Adds pages for managing tax settings and tax rates for regions.
- Fixes `list` tax rates endpoint, which was missing pagination, search, order, and filters.

**Note**
The fix to the tax rate list endpoint is very rough, as I have had to reimplement most of the logic from our transformQuery middleware. This is because this endpoints does not follow normal convention for fields and expand and uses string arrays instead of strings separated by commas. Our middleware does not support this, and changing the endpoint to align with other endpoints on the expand and fields params would be a breaking change. Since this is very temporary until 2.0 I think it's okay for the time being.

CLOSES CORE-1654

### CODE REVIEW

File: packages/admin-next/dashboard/src/routes/taxes/tax-rate-create/components/create-tax-rate-form/create-tax-rate-form.tsx [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+                      />
+                      <Form.Field
+                        control={form.control}
+                        name="code"
+                        render={({ field }) => {
+                          return (
+                            <Form.Item>
+                              <Form.Label>{t("fields.code")}</Form.Label>
+                              <Form.Control>
+                                <Input type="number" {...field} />
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/taxes/tax-rate-create/components/create-tax-rate-form/create-tax-rate-form.tsx]: todo: should be type text

File: packages/admin-next/dashboard/src/routes/taxes/tax-detail/components/tax-general-section/tax-general-section.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            ? t("general.enabled")
+            : t("general.disabled")}
+        </StatusBadge>
+      </div>
+      <div className="text-ui-fg-subtle grid grid-cols-2 items-center px-6 py-4">
+        <Text size="small" weight="plus" leading="compact">
+          {t("fields.taxInclusivePricing")}
+        </Text>
+        <StatusBadge
+          color={region.gift_cards_taxable ? "green" : "grey"}
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/taxes/tax-detail/components/tax-general-section/tax-general-section.tsx]: todo: Should be tax inclusive
```suggestion
          color={region.includes_tax ? "green" : "grey"}
```

**Comment by olivermrbl** [comment]
How do I delete a tax rate?

CleanShot 2024-03-08 at 16 53 42

**Comment by olivermrbl** [comment]
pretty epic work @kasperkristensen, have left some comments 🔥

**Comment by kasperkristensen** [comment]
> How do I delete a tax rate?
> 
> CleanShot 2024-03-08 at 16 53 42

Good question, should be fixed now 😅

**Comment by olivermrbl** [comment]
Still a minor issue with the switches 😅

**Comment by olivermrbl** [comment]
CleanShot 2024-03-08 at 17 15 30

**Pull request merged**